lib/administration/resource/include/irods/resource_administration.hpp

```diff
@@ -41,25 +53,24 @@ namespace irods::experimental::administration
     // Forward declaration for "resource_info" class.
     namespace NAMESPACE_IMPL
     {
-        auto resource_info(RxComm&, const resource_name_type)
-            -> std::tuple<std::error_code, std::optional<class resource_info>>;
+        auto resource_info(RxComm&, const resource_name_type) -> class resource_info;
     } // namespace NAMESPACE_IMPL
 
     /// \brief A namespace defining the default set of resource types.
     namespace resource_type
     {
         // clang-format off
-        const std::string_view compound       = "compound";
-        const std::string_view deferred       = "deferred";
-        const std::string_view load_balanced  = "load_balanced";
-        const std::string_view mockarchive    = "mockarchive";
-        const std::string_view nonblocking    = "nonblocking";
-        const std::string_view passthrough    = "passthru";
-        const std::string_view random         = "random";
-        const std::string_view replication    = "replication";
-        const std::string_view struct_file    = "structfile";
-        const std::string_view universal_mss  = "univmss";
-        const std::string_view unixfilesystem = "unixfilesystem";
+        inline constexpr std::string_view compound       = "compound";
+        inline constexpr std::string_view deferred       = "deferred";
+        inline constexpr std::string_view load_balanced  = "load_balanced";
+        inline constexpr std::string_view mockarchive    = "mockarchive";
+        inline constexpr std::string_view nonblocking    = "nonblocking";
+        inline constexpr std::string_view passthrough    = "passthru";
+        inline constexpr std::string_view random         = "random";
+        inline constexpr std::string_view replication    = "replication";
+        inline constexpr std::string_view struct_file    = "structfile";
+        inline constexpr std::string_view universal_mss  = "univmss";
+        inline constexpr std::string_view unixfilesystem = "unixfilesystem";
```

Comment:
Consider making these `const char* const`.

string_view is great for certain things, but I'm not leaning more to them only being useful as function parameter types and in the impls of functions.

Reply:
Changed to `const char* const` for better flexibility.